### ISSUE TITLE
Bug+Fix: issue with rememberedAudioCategory on iOS

### ISSUE THREAD

**Issue by deJong-IT** [csdcorp/speech_to_text#246]
Before starting the audioSession you remember the last category before changing it:

```
rememberedAudioCategory = self.audioSession.category
try self.audioSession.setCategory(AVAudioSession.Category.playAndRecord, options: .defaultToSpeaker)
```

On deactivating:
```
if let rememberedAudioCategory = rememberedAudioCategory {
    try self.audioSession.setCategory(rememberedAudioCategory)
}
```

But when the options are different (i.e. enable bluetooth, airplay, etc) they will not be remembered and it is possible the current running audio that was on bluetooth is now on the default speaker.

It's better to also remember the options:

```
rememberedAudioCategoryOptions = self.audioSession.categoryOptions
```

```
if let rememberedAudioCategory = rememberedAudioCategory, let rememberedAudioCategoryOptions = rememberedAudioCategoryOptions {
    try self.audioSession.setCategory(rememberedAudioCategory,options: rememberedAudioCategoryOptions)
}
```

**Comment by deJong-IT** [comment]
I created a pullrequest: #249

**Comment by sowens-csd** [comment]
This is available now in 5.0.0, thanks @deJong-IT.
